Source/Core/Logging/RemoteProcessorMetadata+extensions.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  RemoteProcessorMetadata+extensions.swift
+//  Frames
+//
+//  Created by Harry Brown on 03/05/2022.
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import CheckoutEventLoggerKit
+
+extension RemoteProcessorMetadata {
+
+    init(environment: Environment,
+         appPackageName: String,
+         appPackageVersion: String,
+         uiDevice: UIDevice) {
+
+        self.init(productIdentifier: Constants.productName,
+                  productVersion: Constants.version,
+                  environment: environment.rawValue,
+                  appPackageName: appPackageName,
+                  appPackageVersion: appPackageVersion,
+                  deviceName: uiDevice.modelName,
+                  platform: "iOS",
```

Comment:
I think we could also obtain this string from UIDevice, under `systemName`. [Apple Documentation](https://developer.apple.com/documentation/uikit/uidevice/1620054-systemname) doesn't directly clarify possible values, some [SO discussion](https://stackoverflow.com/questions/12038425/possible-values-for-uidevice-currentdevice-systemname) points that it comes as `iOS` from iOS 10+, `iPadOS` from iOS 15+ , as well as `tvOS` and `watchOS` where appropriate.